### PR TITLE
Pre-commit (prettier + black)

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
I thought auto-formatters would help with maintaining the site.

- Added prettier (formats HTML, CSS, JS, YAML, JSON, etc) and black (formats Python) using pre-commit
- Fixed* HTML
- Ran pre-commit

Maintainer TODO:

- add pre-commit ci bot to this repo

---

Feel free to reject this PR if change if too invasive.

*`index.html` has many unclosed tags which I couldn't fix. Added to `.prettier-ignore` for now so that pre-commit doesn't error out.

### CODE REVIEW

**Comment by VeckoTheGecko** [comment]
My html fixes (dd7f3d9) were way to haphazard. I need to re-evaluate

**Comment by erikvansebille** [comment]
Thanks for starting this PR, @VeckoTheGecko. While it looks nice for index.html (and I guess we do need to refactor that; it's gotten a bit out of control!), I'm not sure about all the changes in the other files (e.g. the json files for the drifter data). This red has become somewhat of a catch-all for when we need a lace to put material/data online, and all the reformatting might unintentionally break workflows downstream. 

Also, quite a few files are legacy, e.g. from older conferences such as the vegu21. Not sure they need updating either..

So please don't touch files other than the core ones for the website, in the refactoring?

**Comment by VeckoTheGecko** [comment]
Can do, didn't realise that json data files were in the repo.

Switched the prettier ignore file to an allowlist instead, looking at:

- `utrechtteam.html`
- `faq.html`

Much smaller diff this time 😄

**Comment by erikvansebille** [comment]
OK, this looks much better! But now `index.html` (arguably the most important file) is not updated at all? I see you commented it out of the prettier ignore file; why is it broken?

**Comment by VeckoTheGecko** [comment]
There were quite a few errors within the file (see https://validator.w3.org/nu/?doc=https%3A%2F%2Foceanparcels.org%2F) including tags that weren't closed, and incorrect nesting of elements (e.g. `div`s in `p` tags, which aren't permitted and close the `p` tag early). These errors in particular broke prettier. I was having difficulty hunting these down due to the size of the file and the tooling I was using.

Managed to fix these now I think (with the caveat that the tooling I used in b8301be1271aab13b14f6b2f2fdf3bb6364ae7db made a nasty diff making it difficult to see where the parser inserted the missing closing tags). 09daaba4ff44bc2eec8356c36ea18fe63e57f9c3 used prettier to catch remaining errors (some where beautiful soup misplaced 2 closing tags).

Comparing the final page to the current website, they are visually indistinguishable to me. So I guess the parsing done in b8301be1271aab13b14f6b2f2fdf3bb6364ae7db is the same as how web-browsers parse broken HTML.

This is definitely a messy PR with messy diffs. I'm not even sure if doing these fixes to introduce formatting helps from a maintainence POV, or is just noise. Feel free to close this if its more headache than its worth (no hard feelings from this end) 🙂.

Feel free to specify any [prettier config options](https://prettier.io/docs/en/options.html) you'd like in a [`.prettierrc` json config file](https://prettier.io/docs/en/configuration.html) to customise indentation, max width, etc etc (I saw the index.html file ballooned from 3500 to 9000 lines; partially caused by the hard text wrapping introduced by prettier). If you want, you can let me know the options, and I can cherry pick this PR to ensure that all the files are ran from `master` with the proper options (should only take a few minutes).

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+*
+
+# Ignore all except...
+!index.html
+!utrechtteam.html
+!faq.html
```
Comment (erikvansebille) [inline .prettierignore:6]: Needs an EOL?

**Review by erikvansebille — COMMENTED** [review]
Changes look good! Just one minor comment. And then I guess we (I?) need to set up something local for the pre-commit to work?

**Comment by VeckoTheGecko** [comment]
Either in CI ([pre-commit ci](https://pre-commit.ci/) which is free for open source projects), or local through brew, pypi, or conda ([installation](https://pre-commit.com/#install)). Local setup is just a matter of installing then running `pre-commit install` to install the hooks such that they are run prior to committing. I normally run both local and CI for my projects.

**Pull request merged**